File: Leetcode/1446/1446.cpp
const regex r("([a-z])\\1*");

class Solution {
public:
    int maxPower(string s) {
        int maxValue = 0;
        for (sregex_iterator a(s.begin(), s.end(), r), e; a != e; ++a) {
            maxValue = max(maxValue, (int)a->str().size());
        }
        return maxValue;
    }
};

File: Leetcode/1446/1446.go
func max(a, b int) int {
	if a < b {
		return b
	}
	return a
}

func maxPower(s string) int {
	counter := 0
	var ch byte
	maxVal := 0
	for i := 0; i < len(s); i++ {
		if counter == 0 {
			ch = s[i]
			counter++
			continue
		}
		if s[i] == ch {
			counter++
		} else {
			maxVal = max(maxVal, counter)
			ch = s[i]
			counter = 1
		}
	}
	maxVal = max(maxVal, counter)
	return maxVal
}